src/rest/openapi.yaml

```diff
@@ -43,6 +43,15 @@ paths:
             application/json:
               schema:
                 type: object
+    delete:
+      tags:
+        - node
```

Comment:
It doesn't look accurate that "delete node" means removing the persistent info. I think the first impression may be "delete this device from the Thread network".

What about using  `node/persistent-info`?

Reply:
Hm, in my mind persistent information is like an implementation detail. I don't see us having a `GET` or `PUT` method there.

In a way, logically, we reset/kill the whole (Thread) node, so to me `/node` sound not really wrong... Thoughts?

Reply:
> In a way, logically, we reset/kill the whole (Thread) node, so to me /node sound not really wrong... Thoughts?

My concern is that the method name is "delete", but not "reset", so it could be misleading.

If you are looking for a more generic name, what about "node/network-settings"?

Reply:
> My concern is that the method name is "delete", but not "reset", so it could be misleading.

Granted, `DELETE` is not an ideal method name for this operation, but it is the best fit from what HTTP offers.

> If you are looking for a more generic name, what about "node/network-settings"?

I think I'd rather prefer `node/persistent-info` then.

The thing which bothers me using an endpoint below `/node` is that we delete/reset all the settings of a node, e.g. the `GET` of `/node` or `/node/ext-address` will be different. This, to me, makes the `DELETE` to the parent endpoint `/node` a very good fit.

In the end, its just an endpoint, it doesn't matter too much. I'd still prefer `/node`, but I can live with either endpoint :smile: 

@jwhui do you have thoughts/an opinion on what to choose here?